subprojects/dependency-management/src/main/java/org/gradle/internal/component/external/model/ivy/RealisedIvyModuleResolveMetadata.java

```diff
@@ -90,31 +90,39 @@ private static void addVariantsFromRules(DefaultIvyModuleResolveMetadata compone
         }
 
         for (AdditionalVariant additionalVariant : additionalVariants) {
-            String name = additionalVariant.getName();
-            String baseName = additionalVariant.getBase();
-            ImmutableAttributes attributes;
-            ImmutableCapabilities capabilities;
-            List<ModuleDependencyMetadata> dependencies;
-            ImmutableList<? extends ModuleComponentArtifactMetadata> artifacts;
-
-            ModuleConfigurationMetadata baseConf = (ModuleConfigurationMetadata) declaredConfigurations.get(baseName);
-            if (baseConf == null) {
-                attributes = componentMetadata.getAttributes();
-                capabilities = ImmutableCapabilities.EMPTY;
-                dependencies = ImmutableList.of();
-                artifacts = ImmutableList.of();
-            } else {
-                attributes = baseConf.getAttributes();
-                capabilities = (ImmutableCapabilities) baseConf.getCapabilities();
-                dependencies = Cast.uncheckedCast(baseConf.getDependencies());
-                artifacts = Cast.uncheckedCast(baseConf.getArtifacts());
-            }
+                String name = additionalVariant.getName();
```

Comment:
The whole section is too deep, causing the diff to be too large while you only added code around excludes. Can you fix the formatting?